InsightToolkit: new port InsightToolkit version 5.3.0, replacing InsightToolkit-devel

nilason opened the pull request:
#### Description

<!-- Note: it is best to make pull requests from a branch rather than from master -->

Create new port `InsightToolkit` 5.3.0, replacing `InsightToolkit-devel`. Also add Python subports, along with general Portfile overhaul.


###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
sh -c 'echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion) $(uname -m)"; xcode=$(xcodebuild -version 2>/dev/null); if [ $? == 0 ]; then echo "$(echo "$xcode" | awk '\''NR==1{x=$0}END{print x" "$NF}'\'')"; else echo "Command Line Tools $(pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | awk '\''/version:/ {print $2}'\'')"; fi' | tee /dev/tty | pbcopy
-->
macOS 12.6.6 21G646 arm64
Xcode 14.2 14C18

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried a full install with `sudo port -vst install`?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

nilason commented:
The CI's weren't happy with two conflicting ports, I moved part of this original PR to #20400.

nilason commented:
This depends on merging #20424 .

mascguy commented on graphics/InsightToolkit-devel/Portfile:
```diff
@@ -1,77 +1,161 @@
-master_sites        sourceforge:itk
+homepage            https://itk.org
 
 distname            InsightToolkit-${version}
```
Same comment here, vis-a-vis InsightToolkit4. (In terms of using `dist_subdir InsightToolkit`, to avoid duplication of distribution files.)

mascguy commented:
It might also be worth dropping the Python 3.9 subport, for both InsightToolkit ports.

nilason commented:
> It might also be worth dropping the Python 3.9 subport, for both InsightToolkit ports.

I wouldn't mind. Will do.

I also plan to add to this PR, a port rename. From `InsightToolkit-devel` to just `InsightToolkit` (in the process making the former obsolete). Any thoughts on that?

mascguy commented:
> I also plan to add to this PR, a port rename. From `InsightToolkit-devel` to just `InsightToolkit` (in the process making the former obsolete). Any thoughts on that?

Seems reasonable.

Do you expect this port to remain at major version 5.x? Or is the plan to keep this in-sync with upstream, including updating beyond to 6.x, etc?

nilason commented:
> > I also plan to add to this PR, a port rename. From `InsightToolkit-devel` to just `InsightToolkit ` (in the process making the former obsolete). Any thoughts on that?
> 
> Seems reasonable.
> 
> Do you expect this port to remain at major version 5.x? Or is the plan to keep this in-sync with upstream, including updating beyond to 6.x, etc?

My plan is to keep `InsightToolkit` the latest version available (to continue with 6.x etc). If necessary – for whatever reason  (like the need for `InsightToolkit4` for the `orfeotoolbox` update) – fork and create a `InsightToolkit5` and so forth.

mascguy commented:
> My plan is to keep `InsightToolkit` the latest version available (to continue with 6.x etc). If necessary – for whatever reason  (like the need for `InsightToolkit4` for the `orfeotoolbox` update) – fork and create a `InsightToolkit5` and so forth. 

Sounds great!

nilason commented:
I addressed the issues you had. I also changed this to create a new port `InsightToolkit` and make `InsightToolkit-devel` obsolete. I added myself as co-maintainer for `InsightToolkit`.

Has there ever been a port named `InsightToolkit` and if so, did it have epoch set?

mascguy commented on graphics/InsightToolkit-devel/Portfile:
```diff
@@ -1,77 +1,11 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
```
You'll want to add a comment with the obsolete date.

mascguy commented:
> I addressed the issues you had. I also changed this to create a new port `InsightToolkit` and make `InsightToolkit-devel` obsolete. I added myself as co-maintainer for `InsightToolkit`.
> 
> Has there ever been a port named `InsightToolkit` and if so, did it have epoch set?

There was at one point, but the last version was `3.16.0`. And no, epoch was not set.

mascguy reviewed (approved):
LGTM, apart from obsolete date needed for `InsightToolkit-devel`.

nilason commented:
Obsolete date added.

nilason commented:
> > I addressed the issues you had. I also changed this to create a new port `InsightToolkit` and make `InsightToolkit-devel` obsolete. I added myself as co-maintainer for `InsightToolkit`.
> > Has there ever been a port named `InsightToolkit` and if so, did it have epoch set?
> 
> There was at one point, but the last version was `3.16.0`. And no, epoch was not set.

That explains the "-devel" port for v4+ I suppose.

nilason commented:
Thanks for your assistance!
